rotkehlchen/chain/evm/decoding/hop/decoder.py

```diff
@@ -0,0 +1,259 @@
+                        address=event.address,
+                    )
+                    context.decoded_events.append(fee_event)
+                    maybe_reshuffle_events(
+                        ordered_events=[event, fee_event],
+                        events_list=context.decoded_events,
+                    )
+                break
+        else:
+            # send method works differently than normal bridge events
+            for event in context.decoded_events:
+                if (
+                    event.address == ZERO_ADDRESS and
+                    event.asset == bridge.hop_asset and
+                    event.event_type == HistoryEventType.SPEND and
+                    event.event_subtype == HistoryEventSubType.NONE
+                ):
+                    event.event_type = HistoryEventType.DEPOSIT
+                    event.event_subtype = HistoryEventSubType.BRIDGE
+                    event.counterparty = CPT_HOP
+                    event.notes = f'Bridge {event.balance.amount} {event.asset.symbol_or_name()} to {chain_id.label()} via Hop protocol'  # noqa: E501
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_withdrew(self, context: DecoderContext) -> DecodingOutput:
+        """This function is used to decode the Withdrew event on Hop protocol."""
+        recipient = hex_or_bytes_to_address(context.tx_log.topics[2])
+        if not self.base.is_tracked(recipient):
+            return DEFAULT_DECODING_OUTPUT
+
+        if (bridge := self.bridges.get(context.tx_log.address)) is None:
+            return DEFAULT_DECODING_OUTPUT
+
+        amount_raw = hex_or_bytes_to_int(context.tx_log.data[:32])
+        amount = self._get_amount(amount_raw, bridge)
+
+        for event in context.decoded_events:
+            if (
+                event.address in (bridge.amm_wrapper, bridge.address) and
+                event.asset == bridge.asset and
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.balance.amount == amount
+            ):
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.BRIDGE
+                event.counterparty = CPT_HOP
+                event.notes = f'Bridge {event.balance.amount} {event.asset.symbol_or_name()} via Hop protocol'  # noqa: E501
+                break
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_events(self, context: DecoderContext) -> DecodingOutput:
+        """This function is used to decode the bridging events done via Hop protocol."""
+        if context.tx_log.topics[0] == WITHDRAWAL_BONDED:
+            self._decode_withdrawal_bonded(context=context)
+
+        if context.tx_log.topics[0] == TRANSFER_SENT:
+            self._decode_transfer_sent(context=context)
+
+        if context.tx_log.topics[0] == WITHDREW:
+            self._decode_withdrew(context=context)
```

Comment:
I believe you are missing returns here

```suggestion
            return self._decode_withdrawal_bonded(context=context)

        if context.tx_log.topics[0] == TRANSFER_SENT:
            return self._decode_transfer_sent(context=context)

        if context.tx_log.topics[0] == WITHDREW:
            return self._decode_withdrew(context=context)
```